parse.py

```diff
@@ -93,61 +113,144 @@ def parse_item_info(text: str) -> Item:
+                elif aug.startswith("Item Rarity: "):
+                    iir = int(aug.lstrip("Item Rarity: "))
+                elif aug.startswith("Monster Pack Size: "):
+                    pack_size = int(aug.lstrip("Monster Pack Size: "))
+
+            item_class = Map
         elif first_line == 'Right-click to add this prophecy to your character.':
-            rarity = 'prophecy'
-            return Item(rarity=rarity, name=name)
+            return Prophecy(rarity=rarity, name=name)
         elif first_line.startswith('Can be used in a personal Map Device.'):
-            rarity = 'fragment'  # and scarabs
-            return Item(rarity=rarity, name=name)
+            return Fragment(rarity=rarity, name=name)
         elif first_line.startswith("Combine this with four other different samples in Tane's Laboratory."):
-            rarity = 'metamorph'
-            return Item(rarity=rarity, name=name)  # TODO: metamorph mods and item level
+            return Organ(rarity=rarity, name=name)  # TODO: metamorph mods and item level
         elif first_line.endswith('Right click to drink. Can only hold charges while in belt. Refills as you kill monsters.'):
-            pass  # TODO: handle flasks
-        else:
-            for line in region:
-                mod_value = ','.join(re.findall(r'\d+', line))
-                mod_text = re.sub(r'\d+', '#', line)
-                if mod_text.endswith('(implicit)'):
-                    item_type = ItemModifierType.IMPLICIT
-                    mod_text = mod_text.rstrip(' (implicit)')
-                    mod = all_mods.get((mod_text, item_type))  # get a mod
-                    if mod is not None:
-                        modifiers.append((mod, mod_value))
-                elif mod_text.endswith(' (crafted)'):
-                    item_type = ItemModifierType.CRAFTED
-                    mod_text = mod_text.rstrip(' (crafted)')
-                    mod = all_mods.get((mod_text, item_type))  # get a mod
-                    if mod is not None:
-                        modifiers.append((mod, mod_value))
+            item_class = Flask
+
+    for region in item_list:
+        for line in region:
+            # Match all mods containing (+|-)<number> or The Shaper,
+            # The Elder, etc. These should all be implicits.
+            matches = re.findall(r'([+-]|The )?(\d+|Shaper|Elder|Constrictor|Enslaver|Eradicator|Purifier)', line)
+
+            # First, glue together mods we find in case we encounter + or -
+            # Then, join them into a string separated by ','
+            mod_value = ','.join([''.join(m) for m in matches])
+            mod_text = re.sub(r'([+-]|The )?(\d+|Shaper|Elder|Constrictor|Enslaver|Eradicator|Purifier)', '#', line)
+
+            logging.debug("Parsing %s" % line)
+            if " (implicit)" in mod_text:
+                item_type = ItemModifierType.IMPLICIT
+                mod_text = mod_text[:-11]
```

Comment:
I prefer `mod_text = mod_text.rstrip(' (implicit)')` as it makes it obvious what is happening here instead of slices

Reply:
rstrip removes too much, so we will add comments instead

Reply:
Still haven't commented this in the most previous commit; apologies. I'll get to it asap here.